addons/project_hr_expense/models/project.py

```diff
@@ -68,11 +68,18 @@ def _get_profitability_sequence_per_invoice_type(self):
         sequence_per_invoice_type['expenses'] = 11
         return sequence_per_invoice_type
 
+    def _get_invoice_line_already_included_profitability(self, lines=None):
+        lines = super()._get_invoice_line_already_included_profitability(lines)
+        subset_expense_refund_lines = self.env['account.move.line'].search([('move_type', '=', 'in_refund'), ('expense_id', '!=', False)])
+        expense_refund_move = subset_expense_refund_lines.move_id
+        expense_bill_move = self.env['account.move'].search([('move_type', '=', 'in_invoice'), ('expense_sheet_id', '!=', False)])
+        return super()._get_invoice_line_already_included_profitability(lines) | expense_bill_move.line_ids | expense_refund_move.line_ids
```

Comment:
Why not just selecting the account.move.line linked to an expense? 🤔 
Because here, you will take more lines and some line could not be included in the profitability, right? 🤔 
Also, I think we could have a memory error if we have many `account.move.line` records to fetch. 🙈 
So, it could be better if we just do the search with `_search` to get the ids (since we just need the ids we could return a list of ids too).

Reply:
thanks! I took remark into account, let me know if all good :-)